app/controllers/account/activations_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Account
+  class ActivationsController < ApplicationController
+    before_action :check_user_login
+    skip_before_action :authenticate_user!
+    layout 'guest'
+
+    def new
+      @activation = Activatable::ActivationService.new
+    end
+
+    def update
+      @activation = Activatable::ActivationService.new(params[:activation])
+
+      if @activation.active
+        redirect_to login_path, notice: t('.success')
+      else
+        render :new
+      end
+    end
+
+    def check_user_login
+      redirect_to root_path if user_signed_in?
+    end
+  end
+end
```

Comment:
Bir kullanıcının `activated` alanı `false` olsa bile login olabiliyor, herhangi bir engel yok. Yani önce `Şifremi Unuttum`'a gidip mail adresini girerek bilmediği (çünkü Prospective'den random oluşturuldu) şifresini alabilir, sonra da login olabilir. Öyleyse `activation` bu resimde nerede duruyor?

Reply:
Bu iş aklımda ama kapattığım işler arasında olmadığı için eklemedim. `registrable`'i kaldırırken bunu da halletmem lazım.

Bu işin şöyle bir durumu var:
Bir kişinin hesabının aktifleştirilebilmesi için adaylığının olması gerekiyor. Bir de bizim hiç adaylık oluşturmadan register olduğumuz kayıtlar var. Bu kayıtların toplu olarak aktifleştirilmesi gerekiyor aksi halde sisteme giriş yapamayacaklar. Buraya kadar sıkıntı yok zaten. 

Bundan sonraki aşamada register sayfası kaldırıldığından `student` dışında yeni oluşturulacak kişiler için bu sefer `prospective_employee` üzerinden adaylık oluşturulması gerekecek. Biz henüz `ProspectiveEmployee` tarafını hiç konuşmadık.
Şu an aktifleştirme `prospective_student` baz alınarak yapıldı. Bu işe girdiğim zaman aktifleştirme işlemini her iki adaylığa göre yeniden yazmam gerekecek. Bunu kod içinde bir yere not düşmüştüm zaten.

Reply:
[Şu komit](https://github.com/omu/nokul/pull/895/commits/5f88f2756edc1bf9b88015c75034a58bab0b034f) ile kişi şifresini değiştirse de değiştirmese de hesabını aktifleştirmediği sürece sisteme giriş yapamayacak.